### ISSUE TITLE
Menu: wrong position when using trigger element

### ISSUE THREAD

**Issue by astappiev** [primefaces/primefaces#5035]
## 1) Environment
- PrimeFaces version: 7.0.

An element with `float: right` which is a trigger for the menu. The menu has the following position attributes `my="right bottom" at="right top"`.

## 2) Expected behaviour
Menu positioned by the right side of the element.

## 3) Actual behaviour
The offset position is wrong. The problem in this code
https://github.com/primefaces/primefaces/blob/251ab3b91622d8187ad29b201166019be71350d8/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js#L92-L100

First, it set `display: block` and `visibility: hidden` for the element, and at that point, as the menu container is located outside of content the scrollbar (attached to the `body`) appears (and changes the viewport size). After that, the code calculates the correct location for the element and relocate it, the scrollbar is not needed anymore and it disappears.

As a result, the menu position is not right. We need to add the size of the scroller to offset value of the target element. Or we can calculate position first and then change `display` style.

**Comment by astappiev** [comment]
I have created a patch for myself. Which first do positioning, then changing visibility and it works for me.
```js
PrimeFaces.widget.Menu.prototype.show = function() {
    this.align();
    this.jq.css({
        "z-index": ++PrimeFaces.zindex
    }).show();
};
```

**Comment by melloware** [comment]
@astappiev Thank you for your fix.  I just submitted a PR.  I poked around the entire showcase and tested a few trigger="" and your fix seems to work great with no side effects.

**Comment by astappiev** [comment]
Thanks.

There is another problem exists with positioning when the target element has a decimal width (like `32.75px`, received in result of dividing a sass variable). But it is related to jQuery UI and not PrimeFaces.

Not critical for me, just a known limitation.

**Comment by melloware** [comment]
@astappiev I have attached this reproducer.  This ticket was reported as causing a regression in #6906.

Here is a reproducer you can run with `mvn clean jetty:run` and see that #6906 is broken again.  Can you please update this reproducer or give me an example of your original use case that was not working?  Or else I will have to revert this fix.

Reproducer:
[pf-6906.zip](https://github.com/primefaces/primefaces/files/5888580/pf-6906.zip)

**Comment by astappiev** [comment]
Yes, looks like my "fix" creates regression, I also can see menu detached from modal, but we don't use any menu in modals :(

Here are an example and a reproducer of my issue (it uses the method as it was in PF 7, without my fix, see `test.js`).

![l9JLgteUkC](https://user-images.githubusercontent.com/4512729/106178048-943f5080-6199-11eb-8103-aa85a59234c4.gif)

[pf-example.zip](https://github.com/primefaces/primefaces/files/5889119/pf-example.zip)

**Comment by astappiev** [comment]
First, it may look like that it is because of padding in the button, but if you change or remove it, the menu position is still shifted for 1 rem.

Also with my fix, the position of the menu is sometimes out of the screen, as I [mentioned before](https://github.com/primefaces/primefaces/issues/5035#issuecomment-517256895) it happens when some width is decimal as a result of `calc()` method. Just adjust the width of the screen to fix it, it appears only on small screens as a result of oppened debugger.
![image](https://user-images.githubusercontent.com/4512729/106179378-3e6ba800-619b-11eb-8d6c-3daf5afe2359.png)

**Comment by melloware** [comment]
Yep thanks for the test case let me see if I can fix both use cases.

**Comment by melloware** [comment]
OK latest PR fixes both use cases.

**Comment by melloware** [comment]
@astappiev I think your issue maybe Bootstrap.  Using 10.0.0-SNAPSHOT running your example it happens but I do the same thing in one of my PF Premium Themes it works fine.  So I actually think Bootstrap is somehow causing that extra spacing.

**Comment by astappiev** [comment]
Hi again, it looks we have the issue again after a while.
We weren't using the menu element any more, but today I needed it again, and reproduced this issue (or even #6906 to be more precise).

What happens, is that when I click trigger and before aligning the menu, the scrollbar appears and the menu position ix wrong by the scroll width.

The way I fixed it now, by moving the element to always visible part. Changed this line to `left: '0', top: '0'`
https://github.com/primefaces/primefaces/blob/d12af7291b2496231c9c2aa798a9396a2d12b52f/primefaces/src/main/resources/META-INF/resources/primefaces/menu/menu.base.js#L226

Using `visibility: hidden;` as it was a solution of #6906 doesn't work.
